### ISSUE TITLE
[Bug]:  FTB teams tend to lockup server on spawn boot

### ISSUE THREAD

**Issue by mike3d2008** [FTBTeam/FTB-Mods-Issues#449]
### Mod

FTB Teams

### Mod version

ftb-teams-forge-1802.2.6-build.51

### Forge / Fabric version

forge 40.1.74

### Modpack & version

ATM7 to the sky 1.0.4

### What issue are you having?

were experiencing freezes during boot up at spawn 100% on the server. we had to isolate the snbt data in order for the server to work again, can you look into the files and look at why the server froze and give us a reason, also we do have 7 day force chunk unloading from crash utility enable. this part somewhat familiar to direwolf20 1.18 modpack issue so were thinking its FTB teams

### Crashlogs

none provided but i have linked to the zip files of the snbt data. from world/ftbteam
team https://drive.google.com/file/d/1Pg5ULOD3TYG4dQonZcQ3aXVb_4BfikR7/view?usp=sharing
player https://drive.google.com/file/d/1fLcNmbL5qG-SmHdSs2vjIDOIUUAKGg62/view?usp=sharing

### Steps to reproduce

dont know how or what did it, please review the files

### Anything else to note?

would like some answers

**Comment by mike3d2008** [comment]
In comming server copy with the freeze, please check this full server backup and find the cause of the lock up please so it can be fix, the server member are having fond memory of your Direwolf bug. i just had to wipe both folder in world/ftbteam to get it working again.

https://drive.google.com/file/d/1iPUXjl3AYXS-1zMp6AHtKgdqv7Jz4PZX/view?usp=sharing

**Comment by Raidobw2** [comment]
Hi, sorry for the delay. Thanks for posting the world download, I have managed to try and load it up and it does always freeze at "100%" when loading world.

I attached a VisualVM profiler to it and dumped both the threads and the memory. Unfortunately there is nothing that caught my eye, and no mod was mentioned being blocking in the main server thread, which is usually where the error is. The main server thread was however stuck (it was waiting for something on the thread dump which pointed in the memory dump, but ended up not being useful to look at in that memory address...).

Which folder exactly do you delete in order to boot it up? I could try seeing if there is anything anormal after deleting those files and restarting. I could also check adjacent folders or files to where you are deleting things, plus I have a few other ideas.

Thanks in advance for that information

**Comment by mike3d2008** [comment]
ok, to fix the issue, we had to delete all the files the world/ftbteam , i already told you in the other ticket for direwolf but meh. also we did get help from one of the mods in discord and readjust the restart on our server thinking it was server corruption due to short restart window, it did fix the issue but only delayed it but still keeps happening at 6:30PM on Wednesday. i have the log report to prove that we did everything right. but it still does it. so i dont know, i think something is still bugging it out

https://drive.google.com/file/d/1ut9jwJSMJ6V-kWe3HSe9TyUqCh6RKuVU/view?usp=sharing

**Comment by duckslayr** [comment]
My server is having a similar problem with the same fix in ATM8

**Comment by callmehSpear** [comment]
yes plz fix

**Comment by JustAnotherGhostName** [comment]
This corruption occurs nearly every other restart for my server in 1.19.2 in ATM8 1.0.7. I do have copies of before and after of the ./ServerFiles/world/FTBTeams/ directory, one hour apart from each other. I need to perform further analysis between the files to see what specific data is causing it, but it is a lot of files to go through. I am hoping that using linux tools like diff will help, but regardless for now its practically unusable since a manual intervention is required at most restarts.

**Comment by desht** [comment]
Please attach a zipped copy of the teams data you say prevents your server from starting.  I need to examine it.

**Comment by desht** [comment]
Also, please attach the output of "df" for your Linux server.

**Comment by JustAnotherGhostName** [comment]
> Please attach a zipped copy of the before and after data. I need to examine it.

Certainly, see attached. They are before and after data from 1/8/23 at 3:13 PM US Central and 4:13 PM US Central.
[FTBTeamsGood&Bad.zip](https://github.com/FTBTeam/FTB-Mods-Issues/files/10427104/FTBTeamsGood.Bad.zip)

**Comment by desht** [comment]
before & after:  which is bad and which is good?

**Comment by JustAnotherGhostName** [comment]
Odd, for some reason the comments are not showing on github but I'm getting
the email notification. Regardless, good is before, and bad is after. Good
was taken at 3 PM US Central. bad was taken at 4 PM US Central.

**Comment by JustAnotherGhostName** [comment]
Here is the output. The main drive is higher on usage, but that normally
doesn't reach higher than 90% before files are moved to archives.
[image: image.png]

**Comment by desht** [comment]
Yeah GitHub comments vanished temporarily there, wasn't just you.  I actually unzipped the ftbteamsbad data you sent and dropped it into my test server.  It started just fine and all the player & party data loaded as expected, so that is _not_ the cause of your startup problem, sorry.

As for the differences:
* One new player in the "bad" data (the player's data is fine)
* One player who has the `BypassFTBChunksProtection` flag in the "bad" data, but not the "good" data
* One player who was in a player team in the "good" data, but in a party team in the "bad" data; once again, there party is fine, they obviously just created themselves a party team so they could add an ally.

Once again: your "bad" data is not corrupt, and does not prevent the server from starting.

**Comment by desht** [comment]
And your "df" image is not visible in the comment.

**Comment by desht** [comment]
Nope, Teams doesn't store anything outside the `world/ftbteams` folder.  There's clearly something on your server (that isn't FTB Teams) messing up your startup, because there is nothing wrong with the data you classed as "bad".

**Comment by desht** [comment]
Your decision, of course.  Just pointing out that you don't have bad data now.  I've looked at it and loaded it - it's good.

**Comment by JustAnotherGhostName** [comment]
If it is not a FTB Teams data issue, so be it, but for now if I see anyone report this identical issue, im going to have to advise them to roll back or delete the FTB Teams data.

**Comment by MichaelHillcox** [comment]
As this issue is very old and we've pushed a large amount of changes to the mod. Please check if your issue is still present in 1.21 versions of the mod. If it is, please open a new issue with updated information so it's top of our list 👍

Please do not feel obligated to follow on with this issue, we understand that situations change and we do not expect an update from any of our users.
